plugins/module_utils/github.py

```diff
@@ -232,33 +234,39 @@ def request(
             return json.loads(body)
 
     def paginated_request(self, url, headers=None, timeout=15, **kwargs):
+
         if not url.startswith('http'):
             url = f"{self.gh_url}/{url}"
 
+        result = []
         while url:
             content, response, info = self._request(
                 method='GET',
                 url=url,
                 headers=headers,
                 timeout=timeout,
             )
+            if info["status"] == 404:
+                break
 
             url = get_links(response.headers).get("next", {}).get("url")
+            fetched_items = json.loads(content)
```

Comment:
But I don't remember exactly because it was a while ago. 
I mean I removed the generator implementation because it caused problems and refactoring to a variant with full generator support would have resulted in too many changes which reduced the review perspective. 

If I look at the RSS usage during the execution, the memory requirements are in maximum 70MB and in average 54MB for the entire Ansible Run.
From my perspective that should be okay.

Reply:
As discussed by matrix, i reactivated the iterator now.

I can't easily recreate the original situation because a constellation arose back then where a repo had to be created but the group wasn't there because not everything had been read yet.
Lets see :-)